pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -179,13 +179,13 @@ func buildContainerLogsPath(containerName string, restartCount int) string {
 }
 
 // BuildContainerLogsDirectory builds absolute log directory path for a container in pod.
-func BuildContainerLogsDirectory(podNamespace, podName string, podUID types.UID, containerName string) string {
-	return filepath.Join(BuildPodLogsDirectory(podNamespace, podName, podUID), containerName)
+func BuildContainerLogsDirectory(podLogsDir, podNamespace, podName string, podUID types.UID, containerName string) string {
+	return filepath.Join(BuildPodLogsDirectory(podLogsDir, podNamespace, podName, podUID), containerName)
 }
 
 // BuildPodLogsDirectory builds absolute log directory path for a pod sandbox.
-func BuildPodLogsDirectory(podNamespace, podName string, podUID types.UID) string {
```

Comment:
the method below `parsePodUIDFromLogsDirectory` will fail on custom path. I'm not sure how wide it is being used

Reply:
/lgtm cancel

Reply:
AFAIU [parsePodUIDFromLogsDirectory](https://github.com/kubernetes/kubernetes/blob/674eb36f92dcea33e47ac07d71d88ebe9f5c4c6d/pkg/kubelet/kuberuntime/helpers.go#L197) shouldn't fail with custom path. The only thing it cares about is path ending with `_UID` as it splits by `_` (which will work for both `/var/log/pods/NAMESPACE_NAME_UID` and `/custom_path/NAMESPACE_NAME_UID`).

Reply:
yep, it was me misreading code. Sorry